Files/ViewModels/SettingsViewModels/FilesAndFoldersViewModel.cs

```diff
@@ -0,0 +1,135 @@
+﻿using Microsoft.Toolkit.Mvvm.ComponentModel;
+
+namespace Files.ViewModels.SettingsViewModels
+{
+    public class FilesAndFoldersViewModel : ObservableObject
+    {
+        private bool areHiddenItemsVisible = App.AppSettings.AreHiddenItemsVisible;
+        private bool areSystemItemsHidden = App.AppSettings.AreSystemItemsHidden;
+        private bool showFileExtensions = App.AppSettings.ShowFileExtensions;
+        private bool openItemsWithOneclick = App.AppSettings.OpenItemsWithOneclick;
+        private bool listAndSortDirectoriesAlongsideFiles = App.AppSettings.ListAndSortDirectoriesAlongsideFiles;
+        private bool searchUnindexedItems = App.AppSettings.SearchUnindexedItems;
+
+        public bool AreHiddenItemsVisible
+        {
+            get
+            {
+                return areHiddenItemsVisible;
+            }
+            set
+            {
```

Comment:
Since this class inherits from `ObservableObject` we can drop the check and `OnPropertyChanged()` and replace it with `SetProperty()`

Reply:
As i understand, only Binding is working with DataContext.
![image](https://user-images.githubusercontent.com/20501502/103181445-c06d6600-48a9-11eb-9450-b26966111aa7.png)

Reply:
Yeah, sorry my bad, though it could be solved with `x:Bind` 😅.

Reply:
Like this?
```csharp
if (SetProperty(ref areHiddenItemsVisible, value))
{
    App.AppSettings.AreHiddenItemsVisible = areHiddenItemsVisible;
}

```

Reply:
Yes :100:

Reply:
> `App.AppSettings.AreHiddenItemsVisible = areHiddenItemsVisible;`

`App.AppSettings.AreHiddenItemsVisible = value;`
To make it a bit clearer

Reply:
Thank you!